code_map.py

```diff
@@ -808,22 +762,14 @@ def on_close(self, view):
         if view.file_name() == code_map_file():
 
             if not CodeMapListener.closing_code_map:
-                enabled = settings().get('close_empty_group_on_closing_map', False)
-                if enabled:
-                    # close group only if codemap is the only file in it
-                    
-                    # for some reason the view being closed already disconnected from the window so its group cannot be discovered
-                    # cm_group = win().get_view_index(get_code_map_view())[0]
-                    # alone_in_group = len(win().views_in_group(cm_group)) == 1
-                    
-                    print('closing group with views>', len(win().views_in_group(CodeMapListener.map_group)))
-
-                    alone_in_group = len(win().views_in_group(CodeMapListener.map_group)) == 0
-                    reset_layout(reduce=alone_in_group)    
+                if settings().get('close_empty_group_on_closing_map', False):
+                    sublime.set_timeout_async(lambda: win(
+                        ).run_command('code_map_ensure_group_closed'))
                 return
```

Comment:
the on_close listener wasn't launching the command. Sorry if I stress a bit on this, but it gives problem in certain configurations(including mine). It's related to plugins that close empty groups on their own. In pluginless Sublime that works fine, but with certain plugins it can lead to crashes. Running a delayed window command fixes it in my installation.